streams/src/test/java/org/apache/kafka/streams/integration/KafkaStreamsTelemetryIntegrationTest.java

```diff
@@ -0,0 +1,533 @@
+        }
+
+        @Override
+        public Admin getAdmin(final Map<String, Object> config) {
+            assertTrue((Boolean) config.get(AdminClientConfig.ENABLE_METRICS_PUSH_CONFIG));
+            final TestingMetricsInterceptingAdminClient adminClient = new TestingMetricsInterceptingAdminClient(config);
+            INTERCEPTING_ADMIN_CLIENTS.add(adminClient);
+            return adminClient;
+        }
+    }
+
+    public static class MetricsInterceptingConsumer<K, V> extends KafkaConsumer<K, V> {
+
+        public List<KafkaMetric> passedMetrics = new ArrayList<>();
+
+        public MetricsInterceptingConsumer(final Map<String, Object> configs) {
+            super(configs);
+        }
+
+        public MetricsInterceptingConsumer(final Properties properties) {
+            super(properties);
+        }
+
+        public MetricsInterceptingConsumer(final Properties properties, final Deserializer<K> keyDeserializer, final Deserializer<V> valueDeserializer) {
+            super(properties, keyDeserializer, valueDeserializer);
+        }
+
+        public MetricsInterceptingConsumer(final Map<String, Object> configs, final Deserializer<K> keyDeserializer, final Deserializer<V> valueDeserializer) {
+            super(configs, keyDeserializer, valueDeserializer);
+        }
+
+        @Override
+        public void registerMetricForSubscription(final KafkaMetric metric) {
+            passedMetrics.add(metric);
+            super.registerMetricForSubscription(metric);
+        }
+
+        @Override
+        public void unregisterMetricFromSubscription(final KafkaMetric metric) {
+            passedMetrics.remove(metric);
+            super.unregisterMetricFromSubscription(metric);
+        }
+    }
+
+    public static class TestingClientTelemetry implements ClientTelemetry, MetricsReporter, ClientTelemetryReceiver {
+
+        public TestingClientTelemetry() {
+        }
+
+        @Override
+        public void init(final List<KafkaMetric> metrics) {
+        }
+
+        @Override
+        public void metricChange(final KafkaMetric metric) {
+        }
+
+        @Override
+        public void metricRemoval(final KafkaMetric metric) {
+        }
+
+        @Override
+        public void close() {
+
+        }
+
+        @Override
+        public void configure(final Map<String, ?> configs) {
+
+        }
+
+        @Override
+        public ClientTelemetryReceiver clientReceiver() {
+            return this;
+        }
+
+        @Override
+        public void exportMetrics(final AuthorizableRequestContext context, final ClientTelemetryPayload payload) {
+            try {
+                final Uuid clientId = payload.clientInstanceId();
+                //Temporary until a solution is found for getting shaded MetricData visible to gradle
+                final int dataSize = payload.data().array().length;
+                SUBSCRIBED_CLIENT_METRIC_DATA_COUNT.put(clientId, dataSize);
+
+            } catch (final Exception e) {
```

Comment:
I needed to do this as trying parse the telemetry data with `MetricsData` object from open telemetry is not possible at the moment because gradle can't find the import `org.apache.kafka.shaded.io.opentelemetry.proto.metrics.v1.MetricsData` so until a solution is found I think this is a reasonable proxy for getting a metrics dump.

Reply:
This is resolved now.